lib/v2/shmeea/index.js

```diff
@@ -1,44 +1,49 @@
 const got = require('@/utils/got');
 const cheerio = require('cheerio');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
 
 module.exports = async (ctx) => {
-    const baseURL = 'http://www.shmeea.edu.cn';
-    const rootUrl = baseURL + '/page/08000/index.html';
-    const response = await got({
-        method: 'get',
-        url: rootUrl,
-    });
+    const id = ctx.params.id;
+    const baseURL = 'https://www.shmeea.edu.cn';
+    const link = `${baseURL}/page/${id}/index.html`;
 
-    const data = response.data;
+    const response = await got(link);
+    const $ = cheerio.load(response.data);
 
-    const $ = cheerio.load(data);
+    const title = `上海市教育考试院-${$('#main .pageh4-tit').text().trim()}`;
 
-    const list = $('#main .pageList li');
-
-    const items = await Promise.all(
-        list.map(async (i, item) => {
+    const list = $('#main .pageList li')
+        .toArray()
+        .map((item) => {
             item = $(item);
-            const link = baseURL + item.find('a').attr('href');
-            const description = await ctx.cache.tryGet(link, async () => {
-                const result = await got.get(link);
+            return {
+                title: item.find('a').attr('title') || item.find('a').text(),
+                link: new URL(item.find('a').attr('href'), baseURL).href,
+                pubDate: parseDate(item.find('.listTime').text().trim(), 'YYYY-MM-DD'),
+            };
+        });
 
+    const items = await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const result = await got.get(item.link);
```

Comment:
Please block requests to binary files (pdf,mp4)
https://github.com/DIYgod/RSSHub/actions/runs/7363275233/job/20042569617#step:12:22
https://github.com/DIYgod/RSSHub/actions/runs/7363275233/job/20042569617#step:12:77